### ISSUE TITLE
RuboCop violations highlight

### ISSUE THREAD

**Issue by fullc0ntr0l** [Shopify/vscode-ruby-lsp#732]
### This feature is VS Code specific

- [X] VS Code specific

### Use case

ruby-rubocop gem highlight the first line of a method or class that violates rubocop rules, it's cleaner and easier to read IMO

![image](https://github.com/Shopify/vscode-ruby-lsp/assets/11547199/46cf30a8-6d09-4d10-bbc6-05a3095d32e2)

ruby-lsp highlights the whole method/class and it becomes messy and harder to read for bigger files

![image](https://github.com/Shopify/vscode-ruby-lsp/assets/11547199/33a1d78e-f447-4b3c-9686-fecb9579d2b2)

### Description

Maybe there is such an option to highlight only the first line of the violated block and I missed it, if not it would be cool to have this configurable.

**Comment by andyw8** [comment]
Hi @fullc0ntr0l, 

Is your first screenshot showing another extension in use? Can you provide a link?

**Comment by vinistock** [comment]
This is a duplicate of https://github.com/Shopify/ruby-lsp/issues/355.

TL;DR: it's the responsibility of the linter to decide which AST nodes violations are attached to. The LSP should not have a responsibility to adapt locations that are coming from a linter - especially when our goal is to support multiple linters and formatters.

**Comment by halo** [comment]
> especially when our goal is to support multiple linters

I'm a bit puzzled as to how I can choose another linter, to achieve the desired goal. As far as I can tell, there is no `rubyLsp.linter` setting.

I was able to *additionally add* another linter, simply by installing the [`LoranKloeze.ruby-rubocop-revived`](https://github.com/LoranKloeze/vscode-ruby-rubocop-revived) extension, which works at first:

<img width="172" alt="Screenshot 2023-08-14 at 11 04 39" src="https://github.com/Shopify/vscode-ruby-lsp/assets/11441/911b72c9-a251-49e9-a9bf-2038baded51a">

But after a few seconds, the linter from the RubyLsp extension seems to kick in and override the first lint run:

<img width="147" alt="Screenshot 2023-08-14 at 11 04 06" src="https://github.com/Shopify/vscode-ruby-lsp/assets/11441/6fe6f03d-50f6-4d2d-9d8f-57642b81426d">

To be honest, I've been using vscode for years, but I'm still often confused as to what is configured where (do I maybe have to head over to the `ruby-lsp` gem repository?). I think I expected a "Using linter: none" option here:

<img width="225" alt="Screenshot 2023-08-14 at 11 12 33" src="https://github.com/Shopify/vscode-ruby-lsp/assets/11441/51170086-3f98-47a9-8d27-ead08b7e77b8">

I've not been able to determine at which level the linter output is modified to only highlight one line, in that rubocop-revived gem.

I would appreciate if someone could point me into the right direction. And thank you for Shopify's effort to create some sort of a canonical ruby extension for vscode, it's greatly appreciated.

**Comment by mculp** [comment]
@halo, I’d be interested to know if you made any further progress with a workaround

**Comment by tbuehl** [comment]
@parndt I haven't opened one yet.

**Comment by vinistock** [comment]
@parndt you can just turn off diagnostics in the [language status item](https://github.com/Shopify/vscode-ruby-lsp?tab=readme-ov-file#enable-or-disable-features).

**Comment by parndt** [comment]
> @parndt you can just turn off diagnostics in the [language status item](https://github.com/Shopify/vscode-ruby-lsp?tab=readme-ov-file#enable-or-disable-features).

@vinistock sweet relief, thank you very much. 😌
